File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] res = new int[temperatures.length];
        Stack<Integer> st = new Stack<>();
        for(int i=0; i<temperatures.length; i++) {
            if(!st.isEmpty() && temperatures[st.peek()] >= temperatures[i]) {
                st.push(i);
            }
            else {
                while(!st.isEmpty() && temperatures[st.peek()] < temperatures[i]) {
                    res[st.peek()] = i - st.pop();
                }
                st.push(i);
            }
        }
        return res;
    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        st = deque()
        res = [0 for i in range(len(temperatures))]
        for i in range(len(temperatures)):
            if not st or st and temperatures[st[-1]] >= temperatures[i]:
                st.append(i)
            else:
                while st and temperatures[st[-1]] < temperatures[i]:
                    res[st[-1]] = i - st[-1]
                    st.pop()
                st.append(i)
        return res